packages/rest/src/writer.ts

```diff
@@ -12,21 +12,37 @@ import {Readable} from 'stream';
  *
  * @param response HTTP Response
  * @param result Result from the API to write into HTTP Response
+ * @param contentType Optional content type for the response
  */
 export function writeResultToResponse(
   // not needed and responsibility should be in the sequence.send
   response: Response,
   // result returned back from invoking controller method
   result: OperationRetval,
+  // content type for the response
+  contentType?: string,
 ): void {
-  if (!result) {
-    response.statusCode = 204;
+  // Check if response writing should be bypassed. When the return value
+  // is the response object itself, no writing should happen.
+  if (result === response) return;
+
+  if (result === undefined) {
+    response.status(204);
     response.end();
     return;
   }
 
-  if (result instanceof Readable || typeof result.pipe === 'function') {
-    response.setHeader('Content-Type', 'application/octet-stream');
+  function setContentType(defaultType: string = 'application/json') {
+    if (response.getHeader('Content-Type') == null) {
+      response.setHeader('Content-Type', contentType || defaultType);
+    }
+  }
+
+  if (
+    result instanceof Readable ||
+    typeof (result && result.pipe) === 'function'
```

Comment:
https://loopback.io/doc/en/contrib/style-guide.html#indentation-of-multi-line-expressions-in-if

```js
const isStream = result instanceof Readable || 
  typeof (result && result.pipe) === 'function';

if (isStream) {
  // ...
}
```